### PR TITLE
chore(arithmetic): batch normalize in fuse

### PR DESCRIPTION
**Pull request by TalDerei**
References https://github.com/tachyon-zcash/ragu/issues/545, an obvious _minor_ optimization that hooks into the existing `batch_normalize` utility.

This also adds a criterion bench to measure wall-clock time between sequential and parallel fuse. In thinking about https://github.com/tachyon-zcash/ragu/issues/333, I realized this actually wouldn't offer much benefit because you can only apply this base-point read amortization for MSMs clustered in a _single_ stage, and this only partially benefits some stages. Instead, I tried parallelizing the MSM calls in the prover, but realized our [existing](https://github.com/tachyon-zcash/ragu/pull/510) window-level parallelism within each MSM already likely saturates a normal devices available cores. Halo2 also doesn't implement a multi-core prover (batching MSMs calls) because their prover is seemingly fast enough.

`cargo bench --bench fuse_criterion -p ragu_pcd`
`cargo bench --bench fuse_criterion -p ragu_pcd --features multicore`

-----------

**sequential** 

Screenshot 2026-03-07 at 7 24 21 PM

**Intra-MSM parallelism**

Screenshot 2026-03-08 at 9 12 44 AM

**Inter-MSM parallelism**

Screenshot 2026-03-08 at 9 12 48 AM

### CODE REVIEW

File: crates/ragu_pcd/benches/criterion/fuse.rs [hunk 1]
```diff
@@ -0,0 +1,59 @@
+use criterion::{Criterion, criterion_group, criterion_main};
+use ragu_arithmetic::Cycle;
+use ragu_circuits::polynomials::ProductionRank;
+use ragu_pasta::{Fp, Pasta};
+use ragu_pcd::ApplicationBuilder;
+use ragu_testing::pcd::nontrivial;
+use rand::SeedableRng;
+use rand::rngs::StdRng;
+
+fn fuse_bench(c: &mut Criterion) {
```
Comment (TalDerei) [inline crates/ragu_pcd/benches/criterion/fuse.rs]: same commit from https://github.com/tachyon-zcash/ragu/pull/546

**Review by ebfull — APPROVED** [review]
